common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -112,34 +118,59 @@ public ShuffleIndexInformation load(File file) throws IOException {
     this.errorHandler = new ErrorHandler.BlockPushErrorHandler();
   }
 
+  @VisibleForTesting
+  protected AppShuffleInfo validateAndGetAppShuffleInfo(String appId) {
+    // TODO: [SPARK-33236] Change the message when this service is able to handle NM restart
+    AppShuffleInfo appShuffleInfo =
+      Preconditions.checkNotNull(appsShuffleInfo.get(appId),
```

Comment:
Instead you can change this to `Preconditions.checkArgument` so it doesn't throw NPE and throws IllegalArgumentExceptions. It would be nice if you could check all the places in the class where we can replace the usage of `Preconditions.checkNotNull` with `Preconditions.checkArgument`

Reply:
Updated.